### DISCUSSION TITLE
Detailsuche - Erweiterung der Auswahlliste

### DISCUSSION THREAD

**Discussion by nicole-vlbg** [e-netze/webgis-community#126] [Q&A]
Die Auswahlliste bei der Detailsuche lässt nur 12 Auswahlmöglichkeiten zu.
In diesem ´Fall käme man mit der Eingabe einer Zahl zu weiteren Einträgen. für den Anwender ist das aber nicht unbedingt ersichtlich.
Gibt es hier eine bessere Lösung?

![image](https://github.com/gis-eni/webgis-forum/assets/124671123/3f755084-9f67-46a3-852d-3ddb8bcb3aff)

![image](https://github.com/gis-eni/webgis-forum/assets/124671123/03003348-1de7-4b3d-9bff-7aaa2f2ed5e0)

**Comment by nicole-vlbg** [comment]
...noch eine Ergänzung

Da Einige Cluster rausgefallen sind werden sie nicht durchgängig nummeriert. Da man jetzt nicht alle Cluster sehen kann wie zuvor in Webgis 4 (siehe Bild), weiss der Anwender nicht ob alle vorhanden sind.
![image](https://github.com/gis-eni/webgis-forum/assets/124671123/2f842a48-540f-4cf8-8a4b-8e1330447ce1)

In den Entwicklertools wird das Ergebnis der Abfrage dargestellt – man sieht, dass hier die gesamte Liste abgefragt wird.
![image](https://github.com/gis-eni/webgis-forum/assets/124671123/ace86e94-ebbe-4080-a9d3-7dee024fed8a)
Es wäre für Benutzer hilfreich, wenn sie weitere Inhalte darstellen könnten (und vielleicht auch darauf hingewiesen würden, wie viele Treffer es gibt).

**Comment by jugstalt** [comment]
Zu den "nur" 12 Ergebnissen:
Hab ihr die letzte Version installiert. Hat wir das nicht beim letzten Wednesday auch schon besprochen? Ab jetzt sollte die ersten 200 Treffern in einer scrollbaren Liste angezeigt werden...

**Reply by nicole-vlbg** [reply]
Danke noch für die Antwort, wir haben inzwischen die neue Version installiert. Das funktioniert in der Detailsuche mit den ersten 50 (nicht 200) Treffern in der scrollbaren Liste
Für die Schnellsuche konnten wir uns dann auch wieder erinnern, dass dies konfigurierbar ist.
Hatten wir wohl alles im Zuge unserer WebGIS5 Umstellung verdrängt.

**Reply by faulgurke** [reply]
Ich bitte um Entschuldigung, aber ich habe noch nicht begriffen, wie ich die max. 200 Treffer in der Auswahlliste darstellen kann. Ich schaffe nur 50.
Mir ist klar, dass eine ellenlange Auswahlliste keinen Sinn macht - aber es gibt Datenbereitsteller, die wollen alle ihre Einträge sehen.

**Reply by gis-eni** [reply]
Hast du ein Beispiel für mich, wo nur 50 Ergebnisse angezeigt werden? Mir sind es immer mehr und gibt eigentlich keine Möglichkeit das von WebGIS aus auf 50 zu beschränken?!

**Reply by faulgurke** [reply]
Beispielsweise: 
https://atlas.vorarlberg.at/portal/map/Planung%20und%20Kataster/Kataster%20-%20DKM
Einfach ins Suchfenster "Gemeinde" klicken. Es werden 50 Gemeinden dargestellt - Alberschwende bis Lech (insgesamt sind es 96).

![Detailsuche_Auswahl](https://github.com/gis-eni/webgis-forum/assets/100088983/f41387e2-7915-48b5-87e3-6e9bd81cbde1)

**Reply by gis-eni** [reply]
Sorry, mein Fehler. Hab das Verhalten überprüft und es ist folgendermaßen:
Es werden max 250 Ergebnisse angezeigt.
Das funktioniert allerdings nur, wenn man im CMS für die Auswahlliste eine DB Query mit direktem ConnectionString zur Datenbank angibt. Verwendet man AGS oder gView MapServer Dienste kann man die Abfrage auch direkt über dem Kartenserver machen (was für Auswahllisten auch empfohlen wird). Dazu gibt man als ConnectionString einfach ``#`` an. Da werden dann aber "nur" 50 Ergebnisse abgeholt.
Ich werden das in Zukunft für beide Methoden mit 250 beschränken. Ab Build 5.23.601

Gehe ich recht in der Annahme, dass deine Auswahlliste nicht über einen ConnectionString auf die Datenbank zugreift, sondern über den gView MapServer? Dann wäre das Problem mit dem nächsten Release gelöst.

**Reply by faulgurke** [reply]
Deine Annahme ist korrekt - ich habe leider auch nicht daran gedacht, dass sich direkte DB Zugriffe und Zugriffe über den Service unterschiedlich verhalten könnten. 
Vielen Dank!
